### PR TITLE
chore: adds copilot instructions.

### PR DESCRIPTION
**Pull request by jcchavezs**
We are more often using copilot for reviews and we need to set the instructions to that its catches and suggestions follow our goals and standards.

> Thank you for contributing to Coraza WAF, your effort is greatly appreciated
> Before submitting check if what you want to add to `coraza` list meets [quality standards](https://github.com/corazawaf/coraza/blob/master/CONTRIBUTING.md#quality-standards) before sending pull request. Thanks!

**Make sure that you've checked the boxes below before you submit PR:**

- [ ] My code includes positive and negative tests.
- [ ] I have an appropriate description with correct grammar.
- [ ] I have read [Contribution guidelines](https://github.com/corazawaf/coraza/blob/master/CONTRIBUTING.md#contribution-guidelines), [maintainers note](https://github.com/corazawaf/coraza/blob/master/CONTRIBUTING.md#maintainers) and [Quality standard](https://github.com/github.com/corazawaf/coraza/v3sso/coraza-waf/blob/master/CONTRIBUTING.md#quality-standards).
- [ ] My code is properly linted and passes pre-commit tests.

Thanks for your contribution :heart:

### CODE REVIEW

File: .github/copilot-instructions.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+- Use structured logging
+- Include relevant context (transaction ID, rule ID, etc.)
+- Use appropriate log levels
+- Don't log in tight loops
+- Make logging configurable
+
+## When Adding New Features
+
+1. Come up with a solid use case with more than one potential user
+3. Document the feature
```
Comment (fzipi) [inline .github/copilot-instructions.md]: Should be 2?

File: .github/copilot-instructions.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+- Keep dependencies up to date
+- Use Go modules for dependency management
+
+## Common Patterns
+
+
+### Variable Expansion
+- Support ModSecurity variable syntax (e.g., `%{TX.anomaly_score}`)
+- Handle collections and specific fields
+- Implement proper escaping
```
Comment (fzipi) [inline .github/copilot-instructions.md]: These are the only common patterns?

File: .github/copilot-instructions.md [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+- Prefer channels for communication between goroutines
+- Document thread-safety guarantees
+- Avoid deadlocks by maintaining consistent lock ordering
+- Use `sync.Pool` for object reuse
+
+## File Organization
+
+- Organize code by functionality
+- All packages should be internal unless they are part of the public API
+- Use the experimental/ directory for experimental features
```
Comment (fzipi) [inline .github/copilot-instructions.md]:
```suggestion
- Use the `experimental/` directory for experimental features
```

File: .github/copilot-instructions.md [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+### Testing
+- Write table-driven tests for functions with multiple cases
+- Use subtests with `t.Run()` for logical grouping
+- Mock external dependencies
+- Aim for high test coverage on critical paths
+- Include both positive and negative test cases
+- Test edge cases and error conditions
+
+### Security
+- Never log sensitive data (passwords, tokens, session IDs)
```
Comment (fzipi) [inline .github/copilot-instructions.md:69]: Maybe move this to the `Logging` part?

File: .github/copilot-instructions.md [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+### Performance Considerations
+- **Be obsessed about performance in critical paths and memory leaks**
+- Avoid excessive locking in hot paths
+- Avoid unnecessary allocations in hot paths
+- Use object pooling for frequently created objects (e.g., transactions)
+- Profile and benchmark code before optimizing, focus on real bottlenecks and impactful optimizations (in the order of microseconds rather than nanoseconds)
+- Use efficient data structures (prefer maps for lookups)
+- Minimize the overhead of rule evaluation and only evaluate necessary rules
+
+### Testing
```
Comment (fzipi) [inline .github/copilot-instructions.md:60]: Shall we also prefer using testify?

File: .github/copilot-instructions.md [hunk 6]
```diff
@@ -0,0 +1,136 @@
+# Copilot Instructions for Coraza WAF
+
+## Project Overview
+
+Coraza is a Web Application Firewall (WAF) written in Go that implements the OWASP  Core Rule Set (CRS). It provides protection against common web application attacks.
```
Comment (fzipi) [inline .github/copilot-instructions.md]:
```suggestion
Coraza is a Web Application Firewall (WAF) engine written in Go that implements Seclang directives and it is compatible with OWASP CRS. It provides protection against common web application attacks.
```

**Comment by fzipi** [comment]
Also, why not using a generic `AGENTS.md` file in the top directory instead?

File: .github/copilot-instructions.md (hunk 2 above)
Comment (jcchavezs) [reply .github/copilot-instructions.md]: I wasn't sure what else, maybe I should remove the entire section now and add it later if needed?

File: .github/copilot-instructions.md (hunk 4 above)
Comment (jcchavezs) [reply .github/copilot-instructions.md:69]: Either way works for me.

File: .github/copilot-instructions.md (hunk 5 above)
Comment (jcchavezs) [reply .github/copilot-instructions.md:60]: We are not using testify yet so probably not something we can add ATM.

File: .github/copilot-instructions.md (hunk 2 above)
Comment (fzipi) [reply .github/copilot-instructions.md]: Probably.

File: .github/copilot-instructions.md (hunk 4 above)
Comment (fzipi) [reply .github/copilot-instructions.md:69]: Ok.

**Review by fzipi — APPROVED** [review]

**Comment by jcchavezs** [comment]
> Also, why not using a generic AGENTS.md file in the top directory instead?

Not sure agents is suitable for this but I'd go with this first and then see
